Dshop/apps/users/serializers.py

```diff
@@ -0,0 +1,43 @@
+from rest_framework import serializers
+from rest_framework.serializers import ValidationError
+from django.utils.translation import gettext_lazy as _
+from django.contrib.auth import get_user_model
+
+User = get_user_model()
+
+
+class RegistrationSerializer(serializers.Serializer):
+    """
+    Serializer related to registration of a new user
+    """
+    username = serializers.CharField(min_length=2, max_length=255)
+    email = serializers.EmailField(max_length=255)
+    password1 = serializers.CharField(write_only=True)
+    password2 = serializers.CharField(write_only=True)
+
+    def validate_username(self, username):
+        if User.objects.filter(username=username).exists():
+            raise ValidationError(_('This username already exists'))
+        return username
+
+    def validate_email(self, email):
```

Comment:
Django allows duplicated emails and registration allows it too. Please do not enforce not required constraints.